test/IISExpress.FunctionalTests/InProcess/TestServerTest.cs

```diff
@@ -36,5 +41,162 @@ public TestServerTest(ITestOutputHelper output = null) : base(output)
+
+            Exception exception = null;
+            var cancellationTokenSource = new CancellationTokenSource();
+
+            var data = new byte[1024];
+            using (var testServer = await TestServer.Create(async ctx =>
+            {
+                requestStartedCompletionSource.SetResult(true);
+                try
+                {
+                    while (true)
+                    {
+                        await ctx.Response.Body.WriteAsync(data, cancellationTokenSource.Token);   
+                    }
+                }
+                catch (Exception e)
+                {
+                    exception = e;
+                }
+
+                requestCompletedCompletionSource.SetResult(true);
+            }, LoggerFactory))
+            {
+                using (var connection = testServer.CreateConnection())
+                {
+                    await SendContentLength1Post(connection);
+
+                    await requestStartedCompletionSource.Task.TimeoutAfterDefault();
+                    cancellationTokenSource.Cancel();
+                    await requestCompletedCompletionSource.Task.TimeoutAfterDefault();
+                }
+
+                Assert.IsType<OperationCanceledException>(exception);
+            }
+        }
+
+        [ConditionalFact]
+        public async Task ReaderThrowsCancelledException()
+        {
+            var requestStartedCompletionSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+            var requestCompletedCompletionSource = new TaskCompletionSource<bool>(TaskCreationOptions.RunContinuationsAsynchronously);
+
+            Exception exception = null;
+            var cancellationTokenSource = new CancellationTokenSource();
+
+            var data = new byte[1024];
+            using (var testServer = await TestServer.Create(async ctx =>
+            {
+                requestStartedCompletionSource.SetResult(true);
+                try
+                {
+                    await ctx.Request.Body.ReadAsync(data, cancellationTokenSource.Token);
+                }
+                catch (Exception e)
+                {
+                    exception = e;
+                }
+
+                requestCompletedCompletionSource.SetResult(true);
+            }, LoggerFactory))
+            {
+                using (var connection = testServer.CreateConnection())
+                {
+                    await SendContentLength1Post(connection);
+                    await requestStartedCompletionSource.Task.TimeoutAfterDefault();
+                    cancellationTokenSource.Cancel();
+                    await requestCompletedCompletionSource.Task.TimeoutAfterDefault();
+                }
+                Assert.IsType<OperationCanceledException>(exception);
+            }
+        }
+
+        private static async Task SendContentLength1Post(TestConnection connection)
```

Comment:
The more important trait here is that it includes Connection: Close. Rename method to include that?

Reply:
Nope, the important part is content-length `1` that we never send making ReadAsync block.